src/state_machine.zig

```diff
@@ -409,7 +409,11 @@ pub fn StateMachineType(comptime Storage: type) type {
                     chain = index;
                     assert(chain_broken == false);
                 }
-                const result = if (chain_broken) .linked_event_failed else switch (operation) {
+                const result = if (chain_broken)
+                    .linked_event_failed
+                else if (event.flags.linked and index == events.len - 1)
```

Comment:
We could hide this redundant branch on `event.flags.linked` for performance by refactoring this into the existing `if (event.flags.linked...` branch that we already have on line 408, doing this by means of a `blk` when assigning to `result`.

Reply:
Otherwise, we're adding an extra branch for every event processed by TigerBeetle, even where we're only checking the last event, and only if it's linked. This is also random runtime data so it's hard to predict.

Reply:
Since we already also have the branch that checks for whether an event is linked, we can just fold this cost into that.